common/searchattribute/mapper.go

```diff
@@ -41,13 +41,15 @@ type (
 	}
 )
 
-// ApplyAliases replaces field names with alias names for custom search attributes.
-func ApplyAliases(mapper Mapper, searchAttributes *commonpb.SearchAttributes, namespace string) error {
+// ApplyAliases returns SearchAttributes struct where each search attribute name is replaced with alias.
+// If no replacement where made, it returns nil which means that original SearchAttributes struct should be used.
+func ApplyAliases(mapper Mapper, searchAttributes *commonpb.SearchAttributes, namespace string) (*commonpb.SearchAttributes, error) {
```

Comment:
As long as this is still open...

Would it be possible to rename these to make it obvious that ApplyAliases and SubstituteAliases are inverses? E.g. ApplyAliases/UnapplyAliases or AliasFields/UnaliasFields?